apps/pie-storybook/stories/pie-button.stories.ts

```diff
@@ -120,91 +120,59 @@ const Template = ({
         </pie-button>
         `;
 
-export const Primary: Story<ButtonProps> = (args: ButtonProps) => Template(args);
-Primary.args = {
-    ...defaultArgs,
-};
+interface ButtonStoryOptions extends StoryOptions {
+    variant?: Variant;
+}
 
-export const Secondary: Story<ButtonProps> = (args: ButtonProps) => Template(args);
-Secondary.args = {
-    ...defaultArgs,
+const createStory = <T>(opts?: ButtonStoryOptions) => ({
+    render: (args: T) => Template(args),
+    args: {
+        ...defaultArgs,
+        ...(opts && opts.variant ? { variant: opts.variant } : {}),
```

Comment:
Just wondering if we can make this a bit tidier because we're trying to be prescriptive and generic at the same time

Reply:
Yeh, I was going to abstract this out as a follow-up task. For now, it's slightly button-specific until I move it out into it's own method.

Reply:
I think I'd rather see this function fully generic and typed or save all of it for the follow-up task